x-pack/winlogbeat/module/testing.go

```diff
@@ -0,0 +1,334 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package module
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/stretchr/testify/assert"
+
+	devtools "github.com/elastic/beats/v7/dev-tools/mage"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/esleg/eslegclient"
+	"github.com/elastic/beats/v7/libbeat/mapping"
+	"github.com/elastic/beats/v7/libbeat/version"
+	"github.com/elastic/beats/v7/winlogbeat/module"
+	"github.com/elastic/beats/v7/x-pack/winlogbeat/module/wintest"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
+)
+
+var update = flag.Bool("update", false, "update golden files")
+
+// Option configures the test behavior.
+type Option func(*params)
+
+type params struct {
+	ignoreFields []string
+}
+
+// WithFieldFilter filters the specified fields from the event prior to
+// comparison of values, but retains them in the written golden files.
+func WithFieldFilter(filter []string) Option {
+	return func(p *params) {
+		p.ignoreFields = filter
+	}
+}
+
+// TestIngestPipeline tests the partial pipeline by reading events from the .json files
+// and processing them the ingest pipeline. Then it compares the results against
+// a saved golden file. Use -update to regenerate the golden files.
+func TestIngestPipeline(t *testing.T, pipeline, json string, opts ...Option) {
+	var p params
+	for _, o := range opts {
+		o(&p)
+	}
+	testIngestPipeline(t, pipeline, json, &p)
+}
+
+func testIngestPipeline(t *testing.T, pipeline, pattern string, p *params) {
+	const (
+		host        = "http://localhost:9200"
+		user        = "admin"
+		pass        = "testing"
+		indexPrefix = "winlogbeat-test"
+	)
+
+	paths, err := filepath.Glob(pattern)
+	if err != nil {
+		t.Fatalf("failed to expand glob pattern %q", pattern)
+	}
+	if len(paths) == 0 {
+		t.Fatal("glob", pattern, "didn't match any files")
+	}
+
+	done, _, err := wintest.Docker(".", "test", testing.Verbose())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if *wintest.KeepRunning {
+		fmt.Fprintln(os.Stdout, "docker-compose", "-p", devtools.DockerComposeProjectName(), "rm", "--stop", "--force")
```

Comment:
Assuming I understand the context here, I recommend a message on the line before like "Use this to manually cleanup containers:".